infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/NegotiatingState.java

```diff
@@ -278,6 +278,14 @@ private void processNegotiationResponse(LogReplicationMetadataResponseMsg negoti
         if (negotiationResponse.getSnapshotStart() == negotiationResponse.getSnapshotTransferred() &&
                 negotiationResponse.getSnapshotStart() == negotiationResponse.getSnapshotApplied() &&
                 negotiationResponse.getLastLogEntryTimestamp() >= negotiationResponse.getSnapshotStart()) {
+
+            /*
+             * In the event during the last snapshot sync, apply had processed on the sink, but active had
+             * been interrupted before updating the replication metadata then we should update the prior sync's info.
```

Comment:
it will be useful to call out what 'interrupted' means here - Leadership change, network blip and restart of the leader on active